sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```
@@ -0,0 +1,319 @@
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.METHOD_CALL: {
+                final DetailAST dot = ast.getFirstChild();
+                // method that looks like: method()
+                if (dot.getType() == TokenTypes.IDENT) {
+                    final String methodName = dot.getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+                // method that looks like: obj.method()
+                else {
+                    final String methodName = dot.getLastChild().getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+            }
+            // constructor
+            case TokenTypes.LITERAL_NEW: {
+                if (!includeConstructor) {
+                    break;
+                }
+                final DetailAST constructor = ast.getFirstChild();
+                // case for java8 expression: File::new
+                if (constructor == null) {
+                    break;
+                }
+                final String constructorName = constructor.getText();
+                final DetailAST arguments = ast.findFirstToken(TokenTypes.ELIST);
+                // case for expression: int int[4]
+                if (arguments == null) {
+                    break;
+                }
+                final int numberOfParameters = arguments.getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, constructorName, numberOfParameters);
+                break;
+            }
+            default:
+                break;
```

`default` should throw an exception and we have a test util to do this. Look at our other checks.